GoMeet/src/main/webapp/js/manual-time-proposal.js

```diff
@@ -0,0 +1,161 @@
+ * @param {Document} document the document that this function operates on.
+ * @param {string} inputDivId the id of the div to delete. If invalid or not provided,
+ * this function will do nothing.
+ */
+function deleteTimeInput(document, inputDivId) {
+  if (inputDivId === null) { // No input ID given
+    return;
+  }
+  let inputDiv = document.getElementById(inputDivId);
+  if (inputDiv == null) { // Input ID is invalid (e.g. does not exist)
+    return;
+  }
+  let inputField = inputDiv.children.item(TIME_INPUT_FIELD_INDEX);
+  enteredTimes.delete(inputField.value);
+  inputDiv.remove();
+  timeInputs--; 
+  // Hide delete buttons if they only have one time input left
+  if (timeInputs === 1) {
+    toggleDeleteButtons(document);
+  }
+}
+
+/**
+ * Hides or shows the buttons with className 'delete-time-button'.
+ * Shows buttons if they're hidden, and vice-versa.
+ * Hides buttons by setting buttonElement.style.display to 'none', 
+ * and shows the buttons by setting buttonElement.style.display to
+ * 'inline'.
+ * If no buttons on the page were found with className 'delete-time-button',
+ * this function will do nothing.
+ * @param {Document} document the document that this function operates on.
+ */
+function toggleDeleteButtons(document) {
+  let deleteButtons = document.getElementsByClassName('delete-time-button');
+  for (let i = 0; i < deleteButtons.length; i++) {
+    if (deleteButtons[i].style.display === 'none') {
+      deleteButtons[i].style.display = 'inline';
+    } else {
+      deleteButtons[i].style.display = 'none';
+    }
+  }
+}
+
+/**
+ * Returns the ISO datestring of the local current date and time (hours and minutes),
+ * with the time zone data trimmed.
+ * Seconds and milliseconds are set to 0.
+ * @returns the ISO string in the format YYYY-MM-DDTHH:MM:00:00
+ */
+function getDatetimeNow() {
+  let now = new Date();
+
+  // number of minutes BEHIND that UTC time is relative to local time
+  // e.g. Sydney Time is 660 minutes (11 hrs) ahead, so timezoneOffset = -660
+  let timezoneOffset = now.getTimezoneOffset(); 
+
+  let nowOffset = new Date(now.getTime() - timezoneOffset * 60 * 1000);
+  nowOffset.setMilliseconds(0);
+  nowOffset.setSeconds(0);
+  nowOffset = nowOffset.toISOString();
+  nowOffset = nowOffset.substring(0,nowOffset.length - 1); // trim zone data off
+
+  return nowOffset;
+}
+
+/**
+ * Clears the input field and alerts the user, if the user entered a time that 
+ * is earlier than or equal to the date and time now.
+ * @param {Element} elem the input element where the user inputted the time
+ */
+function rectifyInputtedTime(elem) {
+  let timeEntered = new Date(elem.value);
+  let now = new Date(getDatetimeNow());
+  // If the time entered has been entered before or is earlier than now
+  if (enteredTimes.has(elem.value) || !(timeEntered > now)) {
+    elem.value = '';
+    alert(INVALID_TIME_ERROR);
+  } else {
+    enteredTimes.add(elem.value);
+  }
+}
```

Comment:
Optional: stylistically I think this function could be structured like saveTimes(). particularly:

err = addFutureTime(elem.value);
if (err) {
    alert(err.message)
}

Reply:
For testing, it'll decouple needing to test that err.message gets displayed from validating the behavior of addFutureTime().

Reply:
Yep, I've done that! It does make the tests more focused on testing _either_ the logic of verifying a future time _or_ the UI stuff 😊